apache/metron
# METRON-2023 [UI] Covering Grok Parser Creation with Cypress tests

## Contributor Comments

In this PR I introducing Cypress in Config UI and adding a Cypress test to the Config UI to cover Grok Parser Creation. To do this I made a simplified common test setup across Config and Alert UI to bring them to the same platform.
I also had to remove maven-frontend-plugin to fix issues we faced with on Travis while we tried to add cypress to Config UI project.


The list of changes are the following:
- adding Grok Parser creation test
- removing additional test server solution from Alert UI
- using standard ng serve command to fire up both Alert and Config UI's for testing 
- removing maven-frontend-plugin, excluding UI from maven install and test loops and letting tests run on Travis's own node.js (the issue was deeply investigated and this is the only way to fix issues in the cypress install phase)
- introducing shell script to execute UI tests

## Steps to validate

Please check build logs in Travis CI to see the output of the UI tests.
https://travis-ci.org/apache/metron/jobs/502040547

Under [this link](https://travis-ci.org/tiborm/metron/jobs/502000711) you can find a build failing where I intentionally broke a test to see whether it breaks the build too.


## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  


In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:


### For code changes:
  ```
  mvn -q clean integration-test install && dev-utilities/build-utils/verify_licenses.sh 
  ```


### For documentation related changes:

  ```
  cd site-book
  mvn site
  ```

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.
